support-frontend/assets/components/csr/csrBanner.tsx

```diff
@@ -17,18 +17,23 @@ const banner = css`
 	z-index: 999;
 `;
 
-function CsrBanner() {
-	const [csrCustomerData, setCsrCustomerData] = useState<
-		CsrCustomerData | undefined
-	>(undefined);
+interface CsrBannerProps {
+	csrCustomerData: CsrCustomerData;
+	setCsrCustomerData: ActionCreatorWithPayload<CsrCustomerData>;
+}
+
+function CsrBanner({
+	csrCustomerData,
+	setCsrCustomerData,
+}: CsrBannerProps): JSX.Element | null {
 	useCsrCustomerData(setCsrCustomerData);
 
-	if (csrCustomerData) {
+	if (csrCustomerData.csr.firstName) {
 		return (
 			<div css={container}>
 				<div css={banner}>
 					You are in customer support mode. Signed in as:{' '}
-					{csrUserName(csrCustomerData)}
+					{csrCustomerData.csr.userName ?? csrUserName(csrCustomerData)}
```

Comment:
Is there a reason for checking for two different properties for the CSR user?

Reply:
We use the `csrUserName()` function to construct the csr username from the first and last names then store it in state. The function is just constructing a simple string so I'll remove the additional check